src/Microsoft.AspNet.WebUtilities/BufferedReadStream.cs

```diff
@@ -321,19 +322,15 @@ public string ReadLine(int lengthLimit)
                 ProcessLineChar(builder, ref foundCR, ref foundCRLF);
             }
 
-            if (foundCRLF)
-            {
-                return builder.ToString(0, builder.Length - 2); // Drop the CRLF
-            }
-            // Stream ended with no CRLF.
-            return builder.ToString();
+            return DecodeLine(builder, foundCRLF);
         }
 
         public async Task<string> ReadLineAsync(int lengthLimit, CancellationToken cancellationToken)
         {
             CheckDisposed();
-            StringBuilder builder = new StringBuilder();
+            var builder = new MemoryStream(200);
```

Comment:
is the limitation of 200 enforced elsewhere? Any reason not to just construct with lengthlimit + 1?